SPECS/dracut/20overlayfs/overlayfs-mount.sh

```diff
@@ -0,0 +1,82 @@
+    type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+    VERITY_MOUNT="mnt/verity_mnt"
+    OVERLAY_MOUNT="/mnt/overlay_mnt"
+    OVERLAY_MNT_OPTS="rw,nodev,nosuid,nouser,noexec"
+
+    # Retrieve the verity root. It is expected to be predefined by the dracut cmdline module.
+    [ -z "$root" ] && root=$(getarg root=)
+    # Check if the root variable matches the expected path.
+    # The path "/dev/mapper/root" is hardcoded here because it is a fixed target name
+    # generated by systemd-veritysetup. The name of this dm-verity target is determined
+    # by systemd and cannot be changed, hence the explicit check against this specific path.
+    [[ "$root" == *"/dev/mapper/root"* ]] || die "root mismatches."
+
+    # Retrieve the OverlayFS parameters.
+    [ -z "${overlayfs}" ] && overlayfs=$(getarg rd.overlayfs=)
+    # Retrieve the persistent volume for the OverlayFS.
+    [ -z "${overlayfs_persistent_volume}" ] && overlayfs_persistent_volume=$(getarg rd.overlayfs_persistent_volume=)
+}
+
+# Modified function to mount the physical partition
+mount_physical_partition() {
+    mkdir -p "${OVERLAY_MOUNT}"
+    # Leverage the partition from cmdline
+    local partition="${overlayfs_persistent_volume}"
+
+    if [ -z "${partition}" ]; then
+        # Fallback to volatile overlay if no persistent volume is specified
+        info "No overlayfs persistent volume specified. Creating a volatile overlay."
+        mount -t tmpfs tmpfs -o ${OVERLAY_MNT_OPTS} "${OVERLAY_MOUNT}" || \
+            die "Failed to create overlay tmpfs at ${OVERLAY_MOUNT}"
+    else
+        # Check if /etc/mdadm.conf exists.
+        if [ -f "/etc/mdadm.conf" ]; then
+            mdadm --assemble ${partition} || \
+                die "Failed to assemble RAID volume."
+        fi
+
+        # Mount the specified persistent volume
+        mount "${partition}" "${OVERLAY_MOUNT}" || \
+            die "Failed to mount ${partition} at ${OVERLAY_MOUNT}"
+    fi
+}
+
+create_overlay() {
+    local _dir=$1
+    local _mounted_dir="${VERITY_MOUNT}/${_dir}"
+    local _upper=$2
+    local _work=$3
+
+    [ -d "$_mounted_dir" ] || die "Unable to create overlay as $_dir does not exist"
+
+    mkdir -p "${_upper}" && \
+    mkdir -p "${_work}" && \
+    mount -t overlay overlay -o ro,lowerdir="${_mounted_dir}",upperdir="${_upper}",workdir="${_work}" "${_mounted_dir}" || \
+        die "Failed to mount overlay in ${_mounted_dir}"
+}
+
+mount_root() {
+    info "Mounting DM-Verity Target"
+    mkdir -p "${VERITY_MOUNT}"
+    mount -o ro,defaults "/dev/mapper/root" "${VERITY_MOUNT}" || \
+        die "Failed to mount dm-verity root target"
+
+    mount_physical_partition
+
+    info "Creating OverlayFS"
+    for _group in ${overlayfs}; do
+        IFS=',' read -r overlay upper work <<< "$_group"
+        create_overlay "$overlay" "${OVERLAY_MOUNT}/${upper}" "${OVERLAY_MOUNT}/${work}"
+    done
+
+    info "Done Verity Root Mounting and OverlayFS Mounting"
```

Comment:
where do these `info` message end-up? is it in journal/kmsg? in case of failure, do they show up in the /run/rdsosreport.txt? (can't remember its extension)

Reply:
will they still be in journalctl?
I have seen other dracut modules do other things and I'm not sure what's the right way. For example, the one I've seen is `>> kmsg` after every command it wants its output logged.